src/SoCreate.Extensions.Logging/LoggingMiddleware.cs

```diff
@@ -0,0 +1,97 @@
+using System;
+using System.IO;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Internal;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Serilog.Core;
+using Serilog.Events;
+
+namespace SoCreate.Extensions.Logging
+{
+    public class LoggingMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly ILogger<LoggingMiddleware> _logger;
+        private readonly LoggingLevelSwitch _loggingLevelSwitch;
+        private readonly bool _showTestOutput;
+
+        public LoggingMiddleware(RequestDelegate next, ILogger<LoggingMiddleware> logger,
+            LoggingLevelSwitch loggingLevelSwitch, IConfiguration configuration)
+        {
+            _next = next;
+            _logger = logger;
+            _loggingLevelSwitch = loggingLevelSwitch;
+            _showTestOutput = Convert.ToBoolean(configuration.GetSection("ApplicationProperties").GetSection("ShowTestOutput").Value);
```

Comment:
Move this into the IOptions pattern.  Add options through the extension method when registering logging.